### PR TITLE
d_stage / d_lib

### PR DESCRIPTION
**Pull request by TakaRikka**

### CODE REVIEW

File: src/d/d_stage.cpp [hunk 1]
```diff
@@ -1624,38 +1174,26 @@ asm void dStage_roomControl_c::zoneCountCheck(int param_0) const {
...
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void dStage_roomControl_c::destroyMemoryBlock() {
-    nofralloc
-#include "asm/d/d_stage/destroyMemoryBlock__20dStage_roomControl_cFv.s"
+void dStage_roomControl_c::destroyMemoryBlock() {
+    for (int i = 0; i < 19; i++) {
+        if (mMemoryBlock[i] != NULL) {
+            mDoExt_destroyExpHeap__FP10JKRExpHeap(mMemoryBlock[i]);
```
Comment (jdflyer) [inline src/d/d_stage.cpp:1192]: Why is the function name here mangled?

Comment (TakaRikka) [reply src/d/d_stage.cpp:1192]: was just using the extern in the external refs part. can update the mdoext header and include that instead though

File: src/d/d_stage.cpp [hunk 2]
```diff
@@ -1624,38 +1175,26 @@ asm void dStage_roomControl_c::zoneCountCheck(int param_0) const {
...
-/* 800248A8-8002490C 01F1E8 0064+00 1/1 0/0 0/1 .text destroyMemoryBlock__20dStage_roomControl_cFv
- */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void dStage_roomControl_c::destroyMemoryBlock() {
-    nofralloc
-#include "asm/d/d_stage/destroyMemoryBlock__20dStage_roomControl_cFv.s"
+void dStage_roomControl_c::destroyMemoryBlock() {
+    for (int i = 0; i < 19; i++) {
```
Comment (lepelog) [inline src/d/d_stage.cpp:1191]:
```suggestion
    for (int i = 0; i < ARRAY_SIZE(mMemoryBlock); i++) {
```

File: src/d/d_stage.cpp [hunk 3]
```diff
@@ -1812,53 +1390,50 @@ asm void dStage_roomControl_c::SetTimePass(int param_0) {
...
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void dStage_searchName(char const* param_0) {
-    nofralloc
-#include "asm/d/d_stage/dStage_searchName__FPCc.s"
+dStage_objectNameInf* dStage_searchName(char const* objName) {
+    dStage_objectNameInf* obj = &l_objectName[0];
+
+    for (u32 i = 0; i < 0x38F; i++) {
```
Comment (lepelog) [inline src/d/d_stage.cpp:1402]:
```suggestion
    for (u32 i = 0; i < ARRAY_SIZE(l_objectName); i++) {
```

File: src/d/d_stage.cpp [hunk 4]
```diff
@@ -1812,53 +1390,50 @@ asm void dStage_roomControl_c::SetTimePass(int param_0) {
...
+    char* tmp = NULL;
 
-/* 80024E30-80024EDC 01F770 00AC+00 1/1 0/0 0/0 .text            dStage_getName__FsSc */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-static asm void dStage_getName(s16 param_0, s8 param_1) {
-    nofralloc
-#include "asm/d/d_stage/dStage_getName__FsSc.s"
+    for (int i = 0; i < 0x38F; i++) {
```
Comment (lepelog) [inline src/d/d_stage.cpp:1415]:
```suggestion
    for (int i = 0; i < ARRAY_SIZE(l_objectName); i++) {
```

**Review by lepelog — APPROVED** [review]

**Pull request merged**